apps/nextra/pages/en/build/move/book/functions.mdx

```diff
@@ -609,3 +628,36 @@ There are plans to loosen some of these restrictions in the future, but for now,
 - Avoid marking large functions that are called at different locations as inline. Also avoid inline functions calling lots of other inline functions transitively.
   These may lead to excessive inlining and increase the bytecode size.
 - Inline functions can be useful for returning references to global storage, which non-inline functions cannot do.
+
+## Dot (receiver) function call style
+
+_Since language version 2.0_
+
+By using the well-known name `self` as the first parameter for a function declaration, one can enable calling this function with the `.` syntax -- often also called receiver style syntax. Example:
+
+```move
+module 0x42::example {
+    struct S {}
+
+    fun foo(self: &S, x: u64) { /* ... */ }
+
+    //...
+
+    fun example() {
+        let s = S{};
+        s.foo(1);
+    }
+}
+```
+
+The call `s.foo(1)` is syntactic sugar for `f(&s, 1)`. Notice that the compiler automatically inserts the reference operator. The 2nd, old notation is still available for `f`, so one can incrementally introduce the new call style without breaking existing code.
+
+The type of the `self` argument can be a struct or an immutable or mutable reference to a struct. The struct must be declared in the same module as the function.
+
+Notice that you do not need to `use` the modules which introduce receiver functions. The compiler will find those functions automatically based on the argument type of `s` in a call like `s.foo(1)`. This, in combination with the automatic insertion of reference operators, can make code using this syntax significantly more concise.
+
+
```

Comment:
Seems like a bunch of empty newlines that we would ideally get rid of.

Reply:
Done